### ISSUE TITLE
F-Prop Not running the full length of animations

### ISSUE THREAD

**Issue by patdelaney** [Pixelcontroller/f-prop_issues#4]
I've done my best to try and configure the following scenario with the tool, but the actual behavior is different. I ahev attached to cfg, log and fseq's

**What I'm trying to accomplish:**
I have 2 sequences:
1. Bills_static.fseq  60 sec
2. bills_movement.fseq 30 sec
Both sequences should be loaded into memory.
Sequence 1 should play in a loop at start up
Sequence 2 should play in a loop at button push (immediate Toggle)
Sequence 1 should play in a loop at button push (immediate Toggle)

**What is really happening:**
Power up F-Prop
Sequnce 1 plays in a Loop  (as expected)
Push Button
**Sequence 2 plays, but for only 15 seconds and then Sequence 1 plays for 15 sec**
This looping between the 2 sequences just keeps repeating.

[bills_movement.zip](https://github.com/user-attachments/files/17534134/bills_movement.zip)

**Comment by patdelaney** [comment]
My use case is very similar to the use case in issue #5,
But I would like the first sequence Bills_static.fseq to always be played when device starts.
If a user presses a button it should then play the next sequence (bills_movement.fseq) in a loop until the button is pressed.
When/If the button is pressed again it plays the first sequence in a loop.

**Comment by patdelaney** [comment]
Test Running a config like you suggested for #5.
With the attached config, here is the behavior that I am seeing.
Power up F-prop 
1st sequence plays in a loop (static sequence  30sec)  (no problem, it keeps looping)
SW1 button push
2nd Sequence Bills starts to play. This is a 30sec fseq.
What happens is that it plays for 15 seconds and then it plays the static sequence for 15 seconds. It continues this loop.
If I press the button,depending on which sequnce is playing, it may or may not advnace to the next sequence.
I also thing it doesn't always accept the button push.

[pats_fprop.zip](https://github.com/user-attachments/files/17564106/pats_fprop.zip)

**Comment by keithsw1111** [comment]
I ran your configuration and it seems fine ... see the log below. If you are not getting events using poll that suggests whatever you have connected is not giving a 50ms consistent short to ground ... and thus it is not being detected.

Right now that is what I would be focussed on. I am just testing with a greenie with the ground and sensor 1 shorted and i plug it in to trigger it. As you can see in the log it is working.

[00003701] Falcon Event Player Firmware Build 11 booting.
[00003703] Initialising player mode 0.
[00003705] PM 0 : Play FSEQ.
[00003707] [0] FLASHFSEQInit.
[00003708] Play FLASH FSEQ PM 0; A: 65282.
[00003710] Opening memory file @ 0x10101000 (3993668).
[00003712] Opening memory file @ 0x104d1000 (1996868).
[00003714]     No flash file at index 2.
[00003716]     No flash file at index 3.
[00003717]     No flash file at index 4.
[00003719]     No flash file at index 5.
[00003721]     No flash file at index 6.
[00003722]     No flash file at index 7.
[00003724]     No flash file at index 8.
[00003726]     No flash file at index 9.
[00003728]     No flash file at index 10.
[00003729]     No flash file at index 11.
[00003731]     No flash file at index 12.
[00003733]     No flash file at index 13.
[00003734]     No flash file at index 14.
[00003736]     No flash file at index 15.
[00003738]     No flash file at index 16.
[00003739]     No flash file at index 17.
[00003741]     No flash file at index 18.
[00003743]     No flash file at index 19.
[00003744]     No flash file at index 20.
[00003747]     No flash file at index 21.
[00003748]     No flash file at index 22.
[00003750]     No flash file at index 23.
[00003752]     No flash file at index 24.
[00003753]     No flash file at index 25.
[00003755]     No flash file at index 26.
[00003757]     No flash file at index 27.
[00003758]     No flash file at index 28.
[00003760]     No flash file at index 29.
[00003762]     No flash file at index 30.
[00003763]     No flash file at index 31.
[00003765] Flash contains 2 between 001-032 FSEQ files.
[00003768] Opening memory file @ 0x10101000 (3993668).
[00003770] [0] FLASH Sequential FSEQ between 0 and 0 (2) : 0.
[00003771] [0] FLASHFSEQ: 1.
[00003773] Opening memory file @ 0x10101000 (3993668).
[00003775] [0]    Frame MS: 25, Frames: 2400, Frame Size: 1664, Header Length: 68, Time 60000ms
[00003777] [0]    V2 Format.
[00003779] [0]    Sparse Ranges: 0
[00003780] [0]    Uncompressed.
[00003782] Initialising player mode 1.
[00003784] Initialising sensors.
[00003786] Sensor 1 NO Poll Enabled.
[00003788] File 1: 0x10101000
[00003790] File 2: 0x104d1000
[00003792] **********************************

Event Player running.
[00010088] Sensor 1 NO Poll Triggered.
[00010138] Sensor 1 NO Poll Firing now.
[00010140] Index event running 1.
[00010142] [0] FLASHFSEQNext.
[00010143] Play FLASH FSEQ PM 0; A: 65280.
[00010145] Opening memory file @ 0x10101000 (3993668).
[00010147] Opening memory file @ 0x104d1000 (1996868).
[00010150] [0] FLASH Sequential FSEQ between 0 and 0 (2) : 1.
[00010152] Opening memory file @ 0x10101000 (3993668).
[00010155] [0] FLASHFSEQ: 2.
[00010156] Opening memory file @ 0x104d1000 (1996868).
[00010158] [0]    Frame MS: 25, Frames: 1200, Frame Size: 1664, Header Length: 68, Time 30000ms
[00010165] [0]    V2 Format.
[00010167] [0]    Sparse Ranges: 0
[00010168] [0]    Uncompressed.
[00013212] Sensor 1 NO Poll not triggered ... resetting it.
[00029983] [0] FSEQ is at end.
[00029985] [0] Player mode finished.
[00029987] [0]    Looping.
[00029989] Play FLASH FSEQ PM 0; A: 65282.
[00029990] Opening memory file @ 0x10101000 (3993668).
[00029992] Opening memory file @ 0x104d1000 (1996868).
[00029994] Opening memory file @ 0x10101000 (3993668).
[00029996] [0] FLASHFSEQ: 2.
[00029998] [0]    Frame MS: 25, Frames: 1200, Frame Size: 1664, Header Length: 68, Time 30000ms
[00029999] [0]    V2 Format.
[00030001] [0]    Sparse Ranges: 0
[00030002] [0]    Uncompressed.
[00059987] [0] FSEQ is at end.
[00059989] [0] Player mode finished.
[00059992] [0]    Looping.
[00059993] Play FLASH FSEQ PM 0; A: 65282.
[00059994] Opening memory file @ 0x10101000 (3993668).
[00059996] Opening memory file @ 0x104d1000 (1996868).
[00059998] Opening memory file @ 0x10101000 (3993668).
[00060000] [0] FLASHFSEQ: 2.
[00060002] [0]    Frame MS: 25, Frames: 1200, Frame Size: 1664, Header Length: 68, Time 30000ms
[00060004] [0]    V2 Format.
[00060005] [0]    Sparse Ranges: 0
[00060007] [0]    Uncompressed.
[00068331] Sensor 1 NO Poll Triggered.
[00068381] Sensor 1 NO Poll Firing now.
[00068383] Index event running 1.
[00068384] [0] FLASHFSEQNext.
[00068386] Play FLASH FSEQ PM 0; A: 65280.
[00068387] Opening memory file @ 0x10101000 (3993668).
[00068389] [0] FLASH Sequential FSEQ between 0 and 0 (2) : 0.
[00068391] [0] FLASHFSEQ: 1.
[00068392] Opening memory file @ 0x10101000 (3993668).
[00068394] [0]    Frame MS: 25, Frames: 2400, Frame Size: 1664, Header Length: 68, Time 60000ms
[00068396] [0]    V2 Format.
[00068397] [0]    Sparse Ranges: 0
[00068400] [0]    Uncompressed.
[00071668] Sensor 1 NO Poll not triggered ... resetting it.

**Comment by keithsw1111** [comment]
So i did find a bug where it was not resetting the time when jumping sequence. So if seq 1 was 15 secs in then it would start sequence 2 at 15 seconds if the event triggered. I have sent you a fix for this. Try it out it may solve your issue.

**Comment by patdelaney** [comment]
I loaded it before running into work. I did not have time to pull the logs files, but this is what I observed. 
New firmware loaded, the static sequence (1) ran in a loop,  I pushed the button, it advanced to the 2nd sequence, and played for 15 seconds, not 30, and then looped back to the main sequence for 15 seconds and then back to movement sequence.  it just keeps repeating that. I only pushed the button once.

I'll check the logs when I get home, maybe the new firmware didn't load. I appreciate you looking into this

**Comment by patdelaney** [comment]
Here is the log file. I only pushed the button once and actually pulled the greenie off the board to make sure it wasn't a button issue.

[fep.log](https://github.com/user-attachments/files/17591253/fep.log)

**Comment by keithsw1111** [comment]
And from the log it agrees ...

3.990 - start seq 1
      32.074 - triggered
32.145 - start seq 2
62.128 - start seq 2
92.142 - start seq 2
122.162 - start seq 2
152.176 - start seq 2
182.210 - start seq 2
212.223 - start seq 2
242.237 - start seq 2
272.251 - start seq 2
302.255 - start seq 2
332.308 - start seq 2
362.328 - start seq 2
392.363 - start seq 2
422.377 - start seq 2
452.392 - start seq 2
482.406 - start seq 2
...
22,155.635 - start seq 2

This is exactly what i would expect and there is no evidence in the log that anything ran for 15 seconds. Sequence 2 consistently runs for 30 seconds + a small transition time when it loops. It never switches back. 

We can see it is the right sequence as it has the right duration and it is clear it plays to completion.

**Comment by patdelaney** [comment]
You are correct, it is working as expected, this was human error. I opened up sequence 2 in xlights and played the full sequence. Its a 30 second secuence and the forst 15 sec is the movement effect and then the last 15 seconds is the static effect.

So when I triggered the sequence 2 to play, it did play it in a loop, but I thought it was looping between the sequence 1 and 2, but it was really looping sequence 2. 

I appoligise for all the back and forth, we can close this issue.
